crowbar_framework/lib/crowbar/conduit_resolver.rb

```diff
@@ -34,8 +34,10 @@ def bus_order
         if cr_network_config.key?("interface_map")
           cr_network_config["interface_map"].each do |data|
             next unless cr_dmi_system["product_name"] =~ /#{data["pattern"]}/
-            next if data.key?("serial_number") &&
-                cr_dmi_system["serial_number"].strip != data["serial_number"].strip
+            if data.key?("serial_number")
+              next if data["serial_number"].strip != cr_dmi_system["serial_number"].strip &&
+                  cr_dmi_base_board["serial_number"].strip != data["serial_number"].strip
```

Comment:
the function `cr_dmi_base_board` can return `{}` and then the `strip` will fail:
```
irb(main):008:0> {}['foo'].strip
Traceback (most recent call last):
        2: from /usr/bin/irb:11:in `<main>'
        1: from (irb):8
NoMethodError (undefined method `strip' for nil:NilClass)

```